kv/kvserver: TestFlowControlRaftMembershipV2 failed

kv/kvserver.TestFlowControlRaftMembershipV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/04f1e239-5366-4da2-94b5-df8b3faac071?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.1 @ [8d8afd4d92e7eb4cfaa1af1cb51937ccd8efd15e](https://github.com/cockroachdb/cockroach/commits/8d8afd4d92e7eb4cfaa1af1cb51937ccd8efd15e):


```
=== RUN   TestFlowControlRaftMembershipV2
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFlowControlRaftMembershipV23278012746
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:1057: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlRaftMembershipV23278012746
--- FAIL: TestFlowControlRaftMembershipV2 (77.59s)
=== RUN   TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2
    --- FAIL: TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2 (38.41s)
=== RUN   TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_membership_v2_encoding_apply_to_all:1:
         
        output didn't match expected:
        @@ -152,23 +152,23 @@
           kvflowcontrol.tokens.eval.elastic.deducted                        | 11 MiB   
           kvflowcontrol.tokens.eval.elastic.returned                        | 11 MiB   
           kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.eval.regular.available                       | 80 MiB   
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 11 MiB   
        -  kvflowcontrol.tokens.eval.regular.returned                        | 11 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 10 MiB   
           kvflowcontrol.tokens.eval.regular.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.elastic.available                       | 40 MiB   
           kvflowcontrol.tokens.send.elastic.deducted                        | 11 MiB   
           kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
           kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
           kvflowcontrol.tokens.send.elastic.returned                        | 11 MiB   
           kvflowcontrol.tokens.send.elastic.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.regular.available                       | 80 MiB   
        -  kvflowcontrol.tokens.send.regular.deducted                        | 11 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
           kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        -  kvflowcontrol.tokens.send.regular.returned                        | 11 MiB   
        +  kvflowcontrol.tokens.send.regular.returned                        | 10 MiB   
           kvflowcontrol.tokens.send.regular.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
        --- FAIL: TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (18.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftMembershipV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48071

kv/kvserver.TestFlowControlRaftMembershipV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/949d7d52-c2e5-41e0-88b8-4265d3202536?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.1 @ [96da2472b2c51887c4f0a54ac1523faf61fa4c0b](https://github.com/cockroachdb/cockroach/commits/96da2472b2c51887c4f0a54ac1523faf61fa4c0b):


```
=== RUN   TestFlowControlRaftMembershipV2
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFlowControlRaftMembershipV21064793290
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:1056: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlRaftMembershipV21064793290
--- FAIL: TestFlowControlRaftMembershipV2 (83.72s)
=== RUN   TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2
    --- FAIL: TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2 (43.59s)
=== RUN   TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_membership_v2_encoding_apply_to_all:1:
         
        output didn't match expected:
        @@ -152,23 +152,23 @@
           kvflowcontrol.tokens.eval.elastic.deducted                        | 11 MiB   
           kvflowcontrol.tokens.eval.elastic.returned                        | 11 MiB   
           kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.eval.regular.available                       | 80 MiB   
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 11 MiB   
        -  kvflowcontrol.tokens.eval.regular.returned                        | 11 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 10 MiB   
           kvflowcontrol.tokens.eval.regular.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.elastic.available                       | 40 MiB   
           kvflowcontrol.tokens.send.elastic.deducted                        | 11 MiB   
           kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
           kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
           kvflowcontrol.tokens.send.elastic.returned                        | 11 MiB   
           kvflowcontrol.tokens.send.elastic.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.regular.available                       | 80 MiB   
        -  kvflowcontrol.tokens.send.regular.deducted                        | 11 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
           kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        -  kvflowcontrol.tokens.send.regular.returned                        | 11 MiB   
        +  kvflowcontrol.tokens.send.regular.returned                        | 10 MiB   
           kvflowcontrol.tokens.send.regular.returned.disconnect             | 2.0 MiB  
           kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
        --- FAIL: TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (23.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #147003 kv/kvserver: TestFlowControlRaftMembershipV2 failed [C-test-failure O-robot P-3 T-kv X-duplicate branch-release-25.2.1-rc]
- #146639 kv/kvserver: TestFlowControlRaftMembershipV2 failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-25.2]
- #145298 kv/kvserver: TestFlowControlRaftMembershipV2 failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-25.1.6-rc]
- #144546 kv/kvserver: TestFlowControlRaftMembershipV2 failed [A-testing C-bug C-test-failure O-robot P-3 T-kv X-nostale branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftMembershipV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -35,6 +35,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/echotest"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/admission"
@@ -159,7 +160,6 @@ ORDER BY name ASC;
 func TestFlowControlRangeSplitMergeV2(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TODO(pav-kv): remove when flakes are fixed.
 	defer setRACv2DebugVModule(t)()
 
 	testutils.RunValues(t, "kvadmission.flow_control.mode", []kvflowcontrol.ModeT{
@@ -382,7 +382,6 @@ func TestFlowControlBlockedAdmissionV2(t *testing.T) {
 func TestFlowControlAdmissionPostSplitMergeV2(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TODO(pav-kv): remove when flakes are fixed.
 	defer setRACv2DebugVModule(t)()
 
 	testutils.RunValues(t, "kvadmission.flow_control.mode", []kvflowcontrol.ModeT{
@@ -647,7 +646,6 @@ func TestFlowControlCrashedNodeV2(t *testing.T) {
 func TestFlowControlRaftSnapshotV2(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TODO(#138103): remove when flakes are fixed.
 	defer setRACv2DebugVModule(t)()
 
 	const numServers int = 5
@@ -891,6 +889,9 @@ SELECT store_id,
 func TestFlowControlRaftMembershipV2(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	// And also #148903, #148348, #144546, #141912.
+	skip.UnderDuressWithIssue(t, 149036)
+	defer setRACv2DebugVModule(t)()
 
 	testutils.RunValues(t, "kvadmission.flow_control.mode", []kvflowcontrol.ModeT{
 		kvflowcontrol.ApplyToElastic,
@@ -1023,12 +1024,14 @@ func TestFlowControlRaftMembershipV2(t *testing.T) {
 	})
 }
 
-// TestFlowControlRaftMembershipRemoveSelf tests flow token behavior when the
-// raft leader removes itself from the raft group.
+// TestFlowControlRaftMembershipRemoveSelfV2 tests flow token behavior when
+// the raft leader removes itself from the raft group.
 func TestFlowControlRaftMembershipRemoveSelfV2(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TODO(#137510): remove when flakes are fixed.
+	// Under overload, token deductions may not line up as expected, likely due
+	// to dropping streams.
+	skip.UnderDuressWithIssue(t, 148079)
 	defer setRACv2DebugVModule(t)()
 
 	testutils.RunValues(t, "kvadmission.flow_control.mode", []kvflowcontrol.ModeT{

